File: app/src/main/java/com/datarockets/mnchkn/activities/players/di/PlayersListModule.java
package com.datarockets.mnchkn.activities.players.di;

import com.datarockets.mnchkn.activities.players.PlayersListInteractor;
import com.datarockets.mnchkn.activities.players.PlayersListPresenter;
import com.datarockets.mnchkn.activities.players.PlayersListPresenterImpl;
import com.datarockets.mnchkn.activities.players.PlayersListView;

import dagger.Module;
import dagger.Provides;

@Module
public class PlayersListModule {

    private PlayersListView mPlayersListView;

    public PlayersListModule(PlayersListView playersListView) {
        this.mPlayersListView = playersListView;
    }

    @Provides
    public PlayersListView providesPlayerListView() {
        return mPlayersListView;
    }

    @Provides
    public PlayersListPresenter providesPlayersListPresenter(PlayersListView playersListView,
                                                             PlayersListInteractor interactor) {
        return new PlayersListPresenterImpl(playersListView, interactor);
    }

}

File: app/src/main/kotlin/com/datarockets/mnchkn/activities/players/di/PlayersListModule.kt
package com.datarockets.mnchkn.activities.players.di

import com.datarockets.mnchkn.activities.players.PlayersListInteractor
import com.datarockets.mnchkn.activities.players.PlayersListPresenter
import com.datarockets.mnchkn.activities.players.PlayersListPresenterImpl
import com.datarockets.mnchkn.activities.players.PlayersListView

import dagger.Module
import dagger.Provides

@Module class PlayersListModule(private val mPlayersListView: PlayersListView) {

    @Provides
    fun providesPlayerListView(): PlayersListView {
        return mPlayersListView
    }

    @Provides
    fun providesPlayersListPresenter(playersListView: PlayersListView,
                                     interactor: PlayersListInteractor): PlayersListPresenter {
        return PlayersListPresenterImpl(playersListView, interactor)
    }

}
